### DISCUSSION TITLE
Use subpath in ingress listeners

### DISCUSSION THREAD

**Discussion by maknihamdi** [strimzi/strimzi-kafka-operator#7972] [Q&A] [answered]
When you have only one dns entry (not a wildcard) to your ingress controller, we need to use subpath to forward traffic to your different ingress/services.

We should be able to define a hostname with a subpath for each broker and for bootstrap with ingress listeners.

One alternative is to use internal ClusterIP listener and create "manually" ingresses, with a host and a path, for bootstrap and each broker, but if we want to use ingress lister, we can add new fields "path" to have a full path  like hostname+path

Ex: 
`

          bootstrap:
            host: my-nginx-ingress.mycompany.me
            path: /kafka-cluster-name-bootstrap
          brokers:
            - broker: 0
              host: my-nginx-ingress.mycompany.me
              path: /kafka-cluster-name-broker-0
            - broker: 1
              host: my-nginx-ingress.mycompany.me
              path: /kafka-cluster-name-broker-1
            - broker: 2
              host: my-nginx-ingress.mycompany.me
              path: /kafka-cluster-name-broker-2
`

**Comment by scholzj** [comment, marked as answer]
> One alternative is to use internal ClusterIP listener and create "manually" ingresses, with a host and a path, for bootstrap and each broker, but if we want to use ingress lister, we can add new fields "path" to have a full path like hostname+path

Did you actually tried it? Does it work?

Kafka is not an HTTP protocol. That is why you need to enable TLS passthrough in your Ingress and use it only with TLS. It uses TLS-SNI to pass the connection to the right service. Similarly, the advertised listener configuration in Kafka has place for _path_ since there is nothing like that in the Kafka protocol. So I do not think there is any place for the path on either end and this will simply not work.

That said, if I missed something and you somehow made it work, then please explain how and we can look into it. It would be for sure an interesting addition in such case.

**Reply by maknihamdi** [reply]
Hmmm. You are right
I really didn't know how passthrough works. If it uses sni to resolve service name I dont need to test, it will not work for sure

thank you @scholzj
